src/redis_acme.rs

```diff
@@ -0,0 +1,120 @@
+use crate::acme_store::*;
+use crate::settings::RedisStoreConfig;
+use futures::{future, Future};
+use std::fmt::Display;
+extern crate r2d2_redis;
+use self::r2d2_redis::RedisConnectionManager;
+use self::r2d2_redis::{r2d2, redis};
+
+pub struct RedisAcmeStore {
+  pool: r2d2::Pool<RedisConnectionManager>,
+  ns: Option<String>,
+}
+
+impl RedisAcmeStore {
+  pub fn new(conf: &RedisStoreConfig) -> Self {
+    RedisAcmeStore {
+      pool: r2d2::Pool::new(RedisConnectionManager::new(conf.url.as_str()).unwrap()).unwrap(),
+      ns: conf.namespace.as_ref().cloned(),
+    }
+  }
+
+  fn hostname_key<S: Display>(&self, hostname: S) -> String {
+    format!(
+      "acme-challenge:{}:hostname:{}",
+      self.ns.as_ref().unwrap_or(&"".to_string()),
+      hostname
+    )
```

Comment:
This changes the schema we use to store the challenge's response. It requires knowledge from our web app of which app points to this new platform vs the old one. Is that something we can do / want?

For a while we'll be pointing some apps at this and all the others to the old platform.

We'll have to write to both the old and new keys if we do it this way. That's probably ok btw.

If we're going to do that, we may as well change schema? I was wondering if something like a simple `SET` on  `acme-challenge:<hostname>:<challenge token>` would be better. We can also set a TTL (acme challenges are only valid for a week I think) on it and remove the cleaning up step.

Reply:
I'm glad you asked! I intended on using the same schema but realized we could simplify this quite a bit replacing the sorted set with a simple `SET` and `GET` like you suggested. The TTL will also let us remove the code to prune old challenges on each set. 

What do you think?

Reply:
Let's do it!